lib/Share/RoomShareProvider.php

```diff
@@ -582,46 +582,70 @@ public function getSharesInFolder($userId, Folder $node, $reshares, $shallow = t
 	 */
 	public function getSharesBy($userId, $shareType, $node, $reshares, $limit, $offset): array {
 		$qb = $this->dbConnection->getQueryBuilder();
-		$qb->select('*')
-			->from('share');
+		$qb->select('share.*', 'ta.history_since', 'tr.show_history')
+			->from('share', 'share')
+			->leftJoin('share', 'talk_rooms', 'tr', $qb->expr()->eq('tr.token', 'share.share_with'))
+			->leftJoin('tr', 'talk_attendees', 'ta',
```

Comment:
**Comment:**

This query is executed very often and we are now left joining 2 other tables.
I wonder how big the performance impact is. Maybe we can simulate the query on our instance to measure it a bit.